### PR TITLE
SPA starter monorepo + JSS 22.3.1

### PR DESCRIPTION
**Pull request by addy-pathania**
JSS upgrade 22.3.1
For more info please refer: https://github.com/Sitecore/jss/releases/tag/v22.3.1

### CODE REVIEW

**Review by art-alexeyenko — CHANGES_REQUESTED** [review]
Looks good, some small comments around the containers part.
Did you check with Yavor on whether the container changes are actual and should be in?

File: local-containers/scripts/up.ps1 [hunk 1]
```diff
@@ -15,25 +15,25 @@ $xmCloudHost = $envContent | Where-Object { $_ -imatch "^CM_HOST=.+" }
...
+    $xmCloudClientCredentialsLoginClientId = $envContent | Where-Object { $_ -imatch "^SITECORE_FedAuth_dot_Auth0_dot_ClientCredentialsLogin_ClientId=.+" }
+    $xmCloudClientCredentialsLoginClientSecret = $envContent | Where-Object { $_ -imatch "^SITECORE_FedAuth_dot_Auth0_dot_ClientCredentialsLogin_ClientSecret=.+" }
+    $xmCloudClientCredentialsLoginDomain = $xmCloudClientCredentialsLoginDomain.Split("=")[1]
+    $xmCloudClientCredentialsLoginAudience = $xmCloudClientCredentialsLoginAudience.Split("=")[1]
+    $xmCloudClientCredentialsLoginClientId = $xmCloudClientCredentialsLoginClientId.Split("=")[1]
+    $xmCloudClientCredentialsLoginClientSecret = $xmCloudClientCredentialsLoginClientSecret.Split("=")[1]
 }
 
-#set nuget version
+#set node version
```
[the remark this answers is not in the document]
Comment (addy-pathania) [reply local-containers/scripts/up.ps1:36]: why? its still setting the node version to the env

**Comment by addy-pathania** [comment]
> Looks good, some small comments around the containers part. Did you check with Yavor on whether the container changes are actual and should be in?

Yes I did and he said they are good to go.
cc @yavorsk

File: local-containers/scripts/up.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (art-alexeyenko) [reply local-containers/scripts/up.ps1:36]: the line that does it is
`$nodeVersion = $xmCloudBuild.renderingHosts.nextjsstarter.nodeVersion`
It is very self explanatory without the comment

[the remark this answers is not in the document]
Comment (addy-pathania) [reply local-containers/scripts/up.ps1:36]: The entire block of code is focused on setting the Node.js version, and the comment provides a clear summary of its purpose. While the code may seem self-explanatory, the comment adds clarity, especially for someone new to the script.

[the remark this answers is not in the document]
Comment (art-alexeyenko) [reply local-containers/scripts/up.ps1:36]: I'd rather us remove it for less noise - or expand with something more substantial. Even someone new to the script should be able to deduce that the below four lines of code set a node version

File: local-containers/.env [hunk 2]
```diff
@@ -70,9 +70,9 @@ EXPEDGE_CONNECTION=
 LOG_LEVEL_VALUE=INFO
 
 HOST_LICENSE_FOLDER=C:\License
-NODEJS_VERSION=20.14.0
+NODEJS_VERSION=22.11.0
 VERSION=1.0.1
-JSS_DEPLOYMENT_SECRET_NEXTJS_STARTER=110F1C44A496B45478640DD36F80C18C9
-SITECORE_API_KEY_NEXTJS_STARTER={51DC7C23-A175-4922-B021-282102C4893C}
+JSS_DEPLOYMENT_SECRET_APP_STARTER=110F1C44A496B45478640DD36F80C18C9
+SITECORE_API_KEY_APP_STARTER=
```
[the remark this answers is not in the document]
Comment (yavorsk) [reply local-containers/.env:76]: SITECORE_API_KEY_APP_STARTER is being set by the Init scrip, so we can leave it empty

File: local-containers/scripts/up.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (yavorsk) [reply local-containers/scripts/up.ps1:36]: I noticed that the comment was wrong, so just updated it. My arguments to leave it as it is are:
- the code below it is not just one liner
- this comment feels consistent with the rest comments in the script, so...

**Comment by yavorsk** [comment]
hey @addy-pathania @art-alexeyenko , I confirm the local containers changes should be good. They consist of some variable and local site renamings, that were discussed during the work, no change to functionality

**Review by art-alexeyenko — APPROVED** [review]

**Pull request merged**